src/pages/participants.js

```diff
@@ -4,188 +4,449 @@ import { FaTwitter } from "react-icons/fa";
-            <div className="button is-medium is-link is-in-hero">
-              <span className="icon">
-                <FaTwitter size="fa" />
-              </span>
-              <a
-                href="https://twitter.com/pylondiniumgir1/"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                Keep connected
-              </a>
-            </div>
-          </div>
-        </div>
-      </div>
-    </section>
-    <section className="hero is-primary is-bold is-medium is-hidden-touch is-hidden-tablet">
-      <div className="hero-body">
-        <div className="container">
-          <h1 className="title has-text-centered">The registration is open!</h1>
-        </div>
-      </div>
-    </section>
-    <section className="section is-hidden-touch is-hidden-tablet">
-      <div className="columns is-centered">
-        <div className="column is-half">
+function Request(form) {
+  function validate() {
+    // Validate first name
+    const firstName = formData.get("first_name");
+    inputErrors.firstName = firstName.toLowerCase() === firstName.toUpperCase();
+
+    // Validate last name
+    const lastName = formData.get("last_name");
+    inputErrors.lastName = lastName.toLowerCase() === lastName.toUpperCase();
+
+    // Validate email
+    const emailRegex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/;
+    inputErrors.email = !emailRegex.test(formData.get("email"));
+
+    // It is compulsory to select an os
+    inputErrors.os = formData.getAll("os").length === 0;
+
+    // It is compulsory to select a level of experience
+    const experience = formData.getAll("experience");
+    inputErrors.experience = experience.length === 0;
+
+    // If the experience is not beginner, it is compulsory to say something
+    // about this experience
+    if (experience.filter(elem => elem !== "beginner").length !== 0) {
+      inputErrors.experienceNotes = formData.get("experience_notes") === "";
+    } else {
+      inputErrors.experienceNotes = false;
+    }
+
+    // Check that they have selected only one checkbox in the occupation
+    // field
+    const workingStatus = formData.getAll("working_status");
+    if (workingStatus.length !== 1) {
+      inputErrors.workingStatus = true;
+    } else {
+      // If they have selected "Other", make sure that they are provided
+      // the name of the company.
+      const companyName = formData.get("company_name");
+      if ((workingStatus[0] === "other") && (companyName === "")) {
+        inputErrors.companyName = true;
+      } else {
+        inputErrors.companyName = false;
+      }
+      inputErrors.workingStatus = false;
+    }
+
+    // It is required to explain the motivations for the workshop
+    inputErrors.motivations = formData.get("motivations") === "";
+
+    // It is required to read and accept the coc
+    inputErrors.coc = formData.get("coc") !== "accept";
+
+    for (let errorType in inputErrors) {
```

Comment:
It is customary to add a guard while iterating over objects:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#Iterating_over_the_properties_of_an_object

In this case can be ommitted safely, so just an observation.